src/main/java/com/aws/greengrass/certificatemanager/CertificateManager.java

```diff
@@ -117,10 +119,62 @@ public List<String> getCACertificates() throws KeyStoreException, IOException, C
         return Collections.singletonList(CertificateHelper.toPem(certificateStore.getCACertificate()));
     }
 
+    private X509Certificate[] getX509CACertificates() throws KeyStoreException {
+        return new X509Certificate[]{certificateStore.getCACertificate()};
+    }
+
     public String getCaPassPhrase() {
         return certificateStore.getCaPassphrase();
     }
 
+    /**
+     * Subscribe to server certificate updates.
+     * <p>
+     * The certificate manager will save the given CSR and generate a new certificate under the following scenarios:
+     *   1) The previous certificate is nearing expiry
+     *   2) GGC connectivity information changes
+     * Certificates will continue to be generated until the client calls unsubscribeFromCertificateUpdates.
+     * </p>
+     *
+     * @param getCertificateRequest get certificate request
+     * @throws CertificateGenerationException if unable to generate certificate
+     */
+    public void subscribeToCertificateUpdates(GetCertificateRequest getCertificateRequest)
+            throws CertificateGenerationException {
+        try {
+            GetCertificateRequestOptions.CertificateType certificateType =
+                    getCertificateRequest.getCertificateRequestOptions().getCertificateType();
+            X509Certificate[] caCertificates = getX509CACertificates();
+            KeyPair keyPair = CertificateStore.newRSAKeyPair(2048);
+
+            if (certificateType.equals(GetCertificateRequestOptions.CertificateType.SERVER)) {
+                KeyPair finalKeyPair = keyPair;
+                Consumer<X509Certificate> consumer = (t) -> {
+                    GetCertificateUpdate getCertificateUpdate =
+                            new GetCertificateUpdate(finalKeyPair, t, caCertificates);
```

Comment:
What if the CAs change? Will we be able to take the necessary action?

Reply:
Good catch, that should be in the closure. This is kind of an interim change just to get us off the old APIs, so it will be simplified once we don't have the old consumer type